### PR TITLE
Fix tax total not showing on thank you

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix tax total not showing on thank you

Before
----------------------------------------
image

After
----------------------------------------
image

I also tested with a discount

image

Technical Details
----------------------------------------
This adds support for calling `getSubmittedValue()` & `getSubmittedValues()` from anywhere in the flow, similar to what has been used to simplify the Import forms - but it is only used in this very narrow assign instance

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

File: tests/phpunit/api/v4/Entity/ContributionTest.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace api\v4\Entity;
+
+use api\v4\Api4TestBase;
+use Civi\Test\TransactionalInterface;
+
+/**
+ * @group headless
+ */
+class ContributionTest extends Api4TestBase implements TransactionalInterface {
```
Comment (seamuslee001) [inline tests/phpunit/api/v4/Entity/ContributionTest.php:27]: @eileenmcnaughton this test doesn't seem to relate to the rest of the PR

Comment (eileenmcnaughton) [reply tests/phpunit/api/v4/Entity/ContributionTest.php]: No it doesn't, does it....

**Comment by eileenmcnaughton** [comment]
I fought git & ... I think I won

**Comment by seamuslee001** [comment]
only time can tell

File: CRM/Contribute/Form/ContributionBase.php [hunk 2]
```diff
@@ -1242,6 +1243,48 @@ protected function getExistingContributionID(): ?int {
...
+   * @return mixed|null
+   */
+  public function getSubmittedValue(string $fieldName) {
+    $value = $this->controller->exportValue('Main', $fieldName);
+    if (in_array($fieldName, $this->submittableMoneyFields, TRUE)) {
+      return CRM_Utils_Rule::cleanMoney($value);
+    }
+
+    // Numeric fields are not in submittableMoneyFields (for now)
+    $fieldRules = $this->_rules[$fieldName] ?? [];
```
Comment (seamuslee001) [inline CRM/Contribute/Form/ContributionBase.php:1261]: what is _rules here?

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/ContributionBase.php:1261]: @seamuslee001 yeah - that part would only work on the Main page really - it doesn't matter for the amount implemented so far but I'll tinker with it more in master

**Comment by eileenmcnaughton** [comment]
Some test fixes needed 

CRM_Contribute_Form_Contribution_ThankYouTest::testLiveAndTestContributionStatus
[api_v3_ContributionPageTest.testValidate](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/6405/BKPROF=dfl,SUITES=phpunit-api3,label=bknix-tmp/testReport/junit/(root)/api_v3_ContributionPageTest/testValidate/)
[api_v3_ContributionPageTest.testValidatePost](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/6405/BKPROF=dfl,SUITES=phpunit-api3,label=bknix-tmp/testReport/junit/(root)/api_v3_ContributionPageTest/testValidatePost/)
[api_v3_ContributionPageTest.testValidateOutputOnMissingRecurFields](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/6405/BKPROF=dfl,SUITES=phpunit-api3,label=bknix-tmp/testReport/junit/(root)/api_v3_ContributionPageTest/testValidateOutputOnMissingRecurFields/)

**Comment by eileenmcnaughton** [comment]
This only failed on the e2e unrelated issue

@seamuslee001 @colemanw would be good to get this merged asap because I'm doing lots of UI testing in master to fix another bug & it would give this a really solid r-run (I'm kinda trying to include this in my testing but it keeps 'fallling out' as I flip  between branches)

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
Jenkins says you win!

**Pull request merged**